action/install_gems.rb

```diff
@@ -0,0 +1,65 @@
+class GemfileStrategy
+  def install
+    system("gem install -N #{gem_specifications.join(' ')}")
+  end
+
+  private
+
+  def gem_specifications
+    File
+      .read("Gemfile.lock")
+      .lines
+      .select { |l| line_contains_gem_we_care_about?(l) }
+      .select { |l| line_contains_exact_version?(l) }
+      .map { |l| gemfile_line_to_cli_specification(l) }
+  end
+
+  def gemfile_line_to_cli_specification(line)
+    name, version = line.split(' ')
+    version = version.tr('()', '')
+    %("#{name}:#{version}")
+  end
+
+  def line_contains_gem_we_care_about?(line)
```

Comment:
I'd love to see if there's a way to share this logic between both strategies. Right away, it seems weird that a `Gemfile.lock` project _wouldn't_ install `standard` even though a `gemspec` project would.

Reply:
There was a good reason for the difference...

In theory, we can come across a `.gemspec` that has `standard` listed as a dependency, and does not have `rubocop` listed as a dependency. This gem spec has no explicit mention of `rubocop`, but it will be installed as a dependency's dependency.

In a `Gemfile.lock`, we always have an explicit dependency on `rubocop`, because a `Gemfile.lock` lists out all the dependencies' dependencies. This means we have no need to install `standard`. (`standard` doesn't actually provide any runtime code for `rubocop`, it's restricted to being a convention around `rubocop`'s configuration.)

Now... in practice, if `rubocop` is installed as a dependency of `standard` rather than explicitly, top-level dependency in the `.gemspec`, the `rubocop` command is not available. So a consuming project MUST explicitly depend on `rubocop` in order to be able to run it.

All that is to say, I think we can drop sniffing for `standard` altogether, and put the burden on the consuming repo to explicitly list `rubocop` as a dependency if it wants to use `balto-rubocop`. (That is, unless there is some other way to force ruby gems to make their cli available, even if they're installed as a dependency of something else?)

Reply:
I remembered why I opted to keep `standard` in. `standard` depends on both `rubocop` and `rubocop-performance`. Forcing the consuming repo to list both `rubocop` and `rubocop-performance` is another alternative solution. Do you lean any way in how to resolve this so it's maximally future proof?

Reply:
Personally, I lean towards not requiring the users to modify their dependencies in order to use Balto. I see how it might be a future-proofing concern, but because there is no real material affect on us if a user is locked to an older version (we're not hosting servers running different versions) — it doesn't feel like that big of a concern to me.